src/cursor.rs

```diff
@@ -0,0 +1,190 @@
+use core::fmt;
+use std::{error::Error, sync::Arc};
+
+use crate::platform_impl::PlatformCustomCursor;
+
+/// Use a custom image as a cursor (mouse pointer).
+///
+/// ## Platform-specific
+///
+/// - **Web**
+///     Browsers have limits on cursor sizes. On Chromium, Firefox, and Safari, the maximum
+///     size in 128x128.
+///
+///     On Firefox, [a bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1336764) causes
+///     the cursor to not change until the mouse is moved.
```

Comment:
Actually the specification mandates the cursor to be resized instead of limited: https://www.w3.org/TR/css-ui-3/#ref-for-concrete-object-size:
> If an operating system is **incapable** of rendering a cursor above a given size, cursors larger than that size must be shrunk to within the OS-supported size bounds, while maintaining the cursor image’s intrinsic ratio, if any.

I don't believe Safari actually limits cursor size to 128x128, because I've seen some bug reports of blurry cursors when they are large, but I'm unable to test this myself.
Firefox: https://bugzilla.mozilla.org/show_bug.cgi?id=1253573
Chrome: https://bugs.chromium.org/p/chromium/issues/detail?id=1204322

I don't think we are generally in the habit of documenting browser bugs inside the API documentation. I don't know if we should change that, but we shouldn't do that in this PR. See https://github.com/rust-windowing/winit/issues/2875.

```suggestion
/// **Web**: Some browsers have limits on cursor sizes typically at 128x128.
```

Despite what I said earlier, I think we should still document at least that, because it's a pretty notable bug compared to what we've found so far in the past.

Reply:
Okay thank you

Reply:
But the cursors definitely weren't resized when over 128px, it just showed the default cursor.

Reply:
Do you mean on Safari?

Reply:
I tested on windows, x11 and wayland with chrome and firefox. All of then showed default cursor with over 128px. The exact same happened on macos with safari.

Reply:
Or I think my browsers ran with xwayland but anyway.

Reply:
Yeah Chrome and Firefox are known, I posted the tracking issues of each browser above.
I'm surprised about Safari, FWIW I couldn't find a bug report at WebKit.

Thank you for testing!

Reply:
And I think the blurriness might be because the cursor gets upscaled when using screen scaling. Maybe some mac machines have huge scalings on by default.

Reply:
Ah, that explains it.
Btw, if you have the motivation you could write a bug report at https://bugs.webkit.org similar to https://bugzilla.mozilla.org/show_bug.cgi?id=1253573. Maybe mention the spec as well.

I will actually go ahead and mention the spec in the Firefox issue.

Reply:
I dunno if I care that much :D